osf/migrations/0143_pagecounter_schema.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2018-11-10 18:12
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0142_remove_forks_flag'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='pagecounter',
+            name='action',
+            field=models.CharField(default=b'download', max_length=128, db_index=True),
+        ),
+        migrations.AddField(
+            model_name='pagecounter',
+            name='file',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='pagecounters', to='osf.BaseFileNode'),
+        ),
+        migrations.AddField(
+            model_name='pagecounter',
+            name='guid',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='pagecounters', to='osf.Guid'),
+        ),
+        migrations.AddField(
+            model_name='pagecounter',
+            name='version',
+            field=models.IntegerField(blank=True, null=True, db_index=True),
+        ),
+    ]
```

Comment:
Schema migration - adds action, file, guid, and version fields.

action = charfield, version is integerfield (if null, then this is the sum of all the versions) guid=FK and file= FK.

Made most null=True and blank=True. Do we want to enforce that they're there?

Reply:
Makes sense! I made "resource" formerly guid and file fields required in Part II.